### ISSUE TITLE
Clipping of images when transformations applied

### ISSUE THREAD

**Issue by jrc14** [wieslawsoltes/Svg.Skia#14]
Hi, What's the state of work on drawing images (such as embedded JPEGs) in the library?  I'm seeing a problem where, when I try to render a JPEG image with a transform applied, the image is getting wrongly clipped - I think the clip region is not being transformed.  In the attached SVG, the rectangle gets shown correctly, but the embedded JPEG is messed up.  It looks like it's been transformed correctly, but is then being clipped to the bounds of the original (un-transformed) image:
```SVG
<svg width="100" height="100">
    <rect x="5" y="5" width="40" height="20"
        stroke="green" stroke-width="4" fill="yellow"
        transform="matrix(3 1 -1 3 2 1)"/>          
    <image width="10" height="10" x="0" y="0"
xlink:href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAUAAAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO9TXL0Y4OHwAAAABJRU5ErkJggg=="
        transform="matrix(3 1 -1 3 2 1)"/>
</svg>
```
Have I understood the problem correctly?  Is this something you're planning to work on soon? (it is a feature I need, so I'd be OK to work on it independently, but it might be messy as I am not exactly familiar with how you're doing things).  Thanks.

**Comment by wieslawsoltes** [comment]
I will take a look next week.

wt., 4 lut 2020, 17:15 użytkownik jrc14 <notifications@github.com> napisał:

**Comment by wieslawsoltes** [comment]
Pushed change to current `master`:
![image_clip](https://user-images.githubusercontent.com/2297442/74102217-b3d10980-4b41-11ea-9517-5142401ebc92.png)

```SVG
<svg width="100" height="100" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <rect x="5" y="5" width="40" height="20"
        stroke="green" stroke-width="4" fill="yellow"
        transform="matrix(3 1 -1 3 2 1)"/>          
    <image width="10" height="10" x="0" y="0"
xlink:href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAUAAAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO9TXL0Y4OHwAAAABJRU5ErkJggg=="
        transform="matrix(3 1 -1 3 2 1)"/>
</svg>
```

**Comment by wieslawsoltes** [comment]
Seems like non of the reference svg files has raster image with transform.

**Comment by wieslawsoltes** [comment]
@jrc14 I think this is fixed in master branch:
![image_clip](https://user-images.githubusercontent.com/2297442/74104486-5d6dc600-4b55-11ea-95c6-e2f13e12e94b.png)

**Comment by wieslawsoltes** [comment]
@jrc14 The issue should be fixed in `0.1.1` release.

**Comment by jrc14** [comment]
Thank you for that
